packages/core/src/blocks/types.ts

```diff
@@ -11,6 +11,7 @@ export type BlockItem = {
   searchTerms: string[];
   icon?: JSX.Element;
   render?: (editor: Editor) => JSX.Element | null | true;
+  preview?: string;
```

Comment:
As discussed offline

```suggestion
  preview?: string | () -> React JSX
```